include/swupd.h

```diff
@@ -84,6 +84,7 @@ struct file {
 	int last_change;
 	struct update_stat stat;
 
+	unsigned int fd_valid : 1;
```

Comment:
well.. then use zero. While zero is a valid fd, in this scenario it will never happen for the file you're writing. At least you have the one that is providing you with the data (socket or local file), also the swupd-client is not closing stdout/stderr, which are used to log and communicate with the user.